src/main/scala/cromwell/engine/CallActor.scala

```diff
@@ -1,46 +1,79 @@
 }
 
 /** Actor to manage the execution of a single call. */
-class CallActor(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backend: Backend, workflowDescriptor: WorkflowDescriptor) extends Actor with CromwellActor {
+class CallActor() extends Actor with CromwellActor {
+
+  import CallActor._
 
   type CallOutputs = Map[String, WdlValue]
 
+  val callActorId: UUID = UUID.randomUUID()
+
+  var abortFunction: Option[AbortFunction] = None
+
   private val log = Logging(context.system, classOf[CallActor])
-  val tag = s"CallActor [UUID(${workflowDescriptor.shortId}):${call.name}]"
+  def tag(callReference: CallReference) = s"CallActor [UUID($callActorId)][UUID(${callReference.workflowId})/UUID(${callReference.callId})]"
+  val basicTag = s"CallActor [UUID($callActorId)]"
 
   override def receive = LoggingReceive {
-    case CallActor.Start => handleStart()
-    case CallActor.ExecutionFinished(outputs) => context.parent ! WorkflowActor.CallCompleted(call, outputs)
-    case CallActor.ExecutionFailed(regret) =>
-      log.error(regret, s"$tag: ${regret.getMessage}")
-      context.parent ! WorkflowActor.CallFailed(call, regret.getMessage)
+    case CallActor.Start(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backend: Backend, workflowDescriptor: WorkflowDescriptor) =>
+      handleStart(call, locallyQualifiedInputs, backend, workflowDescriptor)
+    case RegisterCallAbortFunction(callReference: CallReference, abortFunction: AbortFunction) => handleAbortFunction(callReference, abortFunction)
+    case CallActor.ExecutionFinished(callReference, call, outputTry) => handleFinished(callReference, call, outputTry)
+    case CallActor.AbortCall(workflowId: WorkflowID) => handleAbort(workflowId)
     case badMessage =>
-      val diagnostic = s"$tag: unexpected message $badMessage."
+      val diagnostic = s"$basicTag: unexpected message $badMessage."
       log.error(diagnostic)
-      context.parent ! engine.workflow.WorkflowActor.CallFailed(call, diagnostic)
+  }
+
+  private def handleFinished(callReference: CallReference, call: Call, outputTry: Try[CallOutputs]) = {
+    abortFunction = None
+
+    outputTry match {
+      case Success(outputs) => context.parent ! WorkflowActor.CallCompleted(call, outputs)
+      case Failure(e: TaskAbortedException) =>
+            log.info(s"${tag(callReference)}: ${call.name} did not complete because it was aborted.")
+            context.parent ! WorkflowActor.AbortComplete(call)
+      case Failure(e: Throwable) =>
+            log.error(e, s"${tag(callReference)}: ${call.name} failed because ${e.getMessage}")
+            context.parent ! WorkflowActor.CallFailed(call, e.getMessage)
+    }
+  }
+
+  private def handleAbortFunction(callReference: CallReference, abortFunction: AbortFunction) = {
+    this.abortFunction = Some(abortFunction)
+  }
+
+  private def handleAbort(workflowId: WorkflowID) = {
+    abortFunction foreach { _.function()}
```

Comment:
I was recently corrected by the style guide, it should be `{ _.function() }` even though I take issue on this point w/ the guide's author. Sadly, they didn't consult me.